plugins/module_utils/client/discovery.py

```diff
@@ -0,0 +1,160 @@
+# Copyright [2017] [Red Hat, Inc.]
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+import json
+import os
+from collections import defaultdict
+import hashlib
+import tempfile
+
+import kubernetes.dynamic
+import kubernetes.dynamic.discovery
+from kubernetes import __version__
+from kubernetes.dynamic.exceptions import ServiceUnavailableError
+
+from ansible.module_utils import six
+from ansible_collections.kubernetes.core.plugins.module_utils.client.resource import ResourceList
+
+
+class Discoverer(kubernetes.dynamic.discovery.Discoverer):
+    def __init__(self, client, cache_file):
+        self.client = client
+        default_cache_file_name = 'k8srcp-{0}.json'.format(hashlib.sha256(self.__get_default_cache_id()).hexdigest())
+        self.__cache_file = cache_file or os.path.join(tempfile.gettempdir(), default_cache_file_name)
+        self.__init_cache()
+
+    def __get_default_cache_id(self):
+        user = self.__get_user()
+        if user:
+            cache_id = "{0}-{1}".format(self.client.configuration.host, user)
+        else:
+            cache_id = self.client.configuration.host
+
+        if six.PY3:
```

Comment:
Do we need this if we drop py2 support?

Reply:
Nope. I'll remove it. At the time I made this we still had py2 tests being run, but that pr just got merged.